services/wireguard/service.go

```diff
@@ -96,7 +123,7 @@ func (manager *Manager) Start(providerID identity.Identity) (dto_discovery.Servi
 	return proposal, sessionConfigProvider, nil
 }
 
-// Wait blocks until service is stopped
+// Wait blocks until service is stopped.
 func (manager *Manager) Wait() error {
 	if !manager.isStarted {
```

Comment:
do we really need these "isStarted" checks? We assume services have a clear lifecycle already defined - create, start (stop in paralle) wait for return

Reply:
`isStarted` here just to support `Wait` method. Otherwise, we will need to periodically check if wireguard interface active and sleep until next check.

Reply:
It can be expressed easily with WaitGroup -> start wg.Add(1), stop -> cleanup and then wg.Done() , wait -> wg.Wait(). No ifs

Reply:
Left only WaitGroup here.